test/test_torch.py

```diff
@@ -8828,6 +8828,14 @@ def gen_nontrivial_input(num_src, dtype, device):
                             self.assertNotEqual(tensor[dst1[i, 0], dst1[i, 1], dst1[i, 2]].item(), 0)
                         lex = is_lexicographically_sorted(dst1)
                         self.assertEqual(torch.ones_like(lex), lex)
+                    if TEST_NUMPY:
```

Comment:
This needs a test for the empty tensor (can be added to `test_nonzero_empty` below). It also needs a test for a zero-dim (scalar) tensor (i.e. `torch.tensor(1)` and `torch.tensor(0)`).

Reply:
Case that seems to be different than numpy:
```
>>> torch.nonzero_tuple(torch.tensor(0))
()

>>> torch.nonzero_tuple(torch.tensor(5))
()

>>> np.array(5).nonzero()
(array([0]),)

>>> np.array(0).nonzero()
(array([], dtype=int64),)
```

Reply:
I think this is a bug in numpy. It doesn't match their documentation: "Returns a tuple of arrays, **one for each dimension of a**" ... I think it's semantically correct to do it the way I have it now, but I can see the argument for bug-for-bug compatibility with np... what do you guys think?

Reply:
I think we should keep numpy behavior.
When performing advanced indexing with a scalar boolean tensor containing True, it should return the same as if we passed bool True, which returns all the dimensions.

Reply:
@fmassa I'm not sure how to interpret your comment.

How is boolean indexing relevant here?

Doesn't `True` insert a dimension?

Reply:
@gchanan scratch what I just said. Upon checking [the numpy detailed notes of advanced indexing](https://www.numpy.org/devdocs/reference/arrays.indexing.html#detailed-notes)
> the nonzero equivalence for Boolean arrays does not hold for zero dimensional boolean arrays

So what I said doesn't hold.

---
For completeness, I was trying to make a point that the following should hold:
```python
x = a[np.array(True)]
y = a[True]
z = a[np.array(True).nonzero()]
# this is true
x == y
# but z is handled differently by numpy
x != z
```